### ISSUE TITLE
PAL SleepEx tests are problematic

### ISSUE THREAD

**Issue by pgavlin** [dotnet/coreclr#10261]
These tests depend on the PAL's implementation of `SleepEx` returning within a certain amount of time. As far as I am aware, `pthread_cond_timedwait` makes no guarantees that it will return within a particular amount of time, only that it will not return before its timeout expires.

See https://github.com/dotnet/coreclr/pull/10239 for an example of some CI failures due to these tests.

**Comment by pgavlin** [comment]
cc @janvorli

**Comment by kouvel** [comment]
This is a general problem with all of the wait tests that verify that a wait completes in reasonable time, especially when the machine is overloaded running other tests in parallel. These tests are useful when bringing up new systems and a sufficient buffer should be included when expecting wait completion. The buffer needs to be adjusted, for that part I'll consider this a duplicate of #8321.
